vyper/semantics/analysis/annotation.py

```diff
@@ -231,10 +231,13 @@ def visit_Subscript(self, node, type_):
 
             elif type_ is not None and len(possible_base_types) > 1:
                 for possible_type in possible_base_types:
-                    if isinstance(possible_type.value_type, type(type_)):
+                    if type_.compare_type(possible_type.value_type):
                         base_type = possible_type
                         break
 
+                else:
+                    raise TypeMismatch(f"Expected {type_} but it is not a possible type", node)
```

Comment:
TypeMismatch is not a panic! either `CompilerPanic` or `TypeCheckFailure` would be appropriate here.

Reply:
added in https://github.com/vyperlang/vyper/pull/3378/commits/2cb947ad626da99ebabc35e37d9dda06cca6792d